docs/DeveloperGuide.md

```diff
@@ -6,17 +6,146 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+Our program is structured as pictured in the following class diagram:
+![ClassDiagram.png](diagrams/ClassDiagram.png)
```

Comment:
Can consider having segments in the DG that goes into more details about the different classes such as `Parser`, `Ui` and `ModuleList` so that reader can understand what they do in the context of your application.